### ISSUE TITLE
Composer update fails on 404

### ISSUE THREAD

**Issue by webmake** [composer/composer#10603]
Hello, it became impossible to update packages. we don't have direct dependency, to react/promise, which is in lock:
```
            "name": "react/promise",
            "version": "v2.9.0",
```
but composer update fails like following:
```
Reading /var/composer/cache/repo/https---repo.packagist.org/provider-symfony~var-dumper~dev.json from cache
Reading /var/composer/cache/repo/https---repo.packagist.org/provider-symfony~web-profiler-bundle~dev.json from cache
[304] https://repo.packagist.org/p2/symfony/config.json
[304] https://repo.packagist.org/p2/symfony/dependency-injection.json
[304] https://repo.packagist.org/p2/symfony/http-kernel.json
[304] https://repo.packagist.org/p2/symfony/cache.json
[304] https://repo.packagist.org/p2/symfony/console.json
Downloading https://repo.packagist.org/p2/php-http/client-implementation.json
Downloading https://repo.packagist.org/p2/symfony/filesystem.json if modified
Downloading https://repo.packagist.org/p2/symfony/polyfill-mbstring.json if modified
Downloading https://repo.packagist.org/p2/symfony/polyfill-intl-idn.json if modified
Downloading https://repo.packagist.org/p2/symfony/security-bundle.json if modified
[304] https://repo.packagist.org/p2/symfony/deprecation-contracts.json
[304] https://repo.packagist.org/p2/symfony/doctrine-bridge.json
[304] https://repo.packagist.org/p2/symfony/framework-bundle.json
[304] https://repo.packagist.org/p2/symfony/service-contracts.json
[304] https://repo.packagist.org/p2/symfony/stopwatch.json
Downloading https://repo.packagist.org/p2/symfony/serializer.json if modified
Downloading https://repo.packagist.org/p2/symfony/yaml.json if modified
Downloading https://repo.packagist.org/p2/symfony/monolog-bundle.json if modified
Downloading https://repo.packagist.org/p2/symfony/proxy-manager-bridge.json if modified
Downloading https://repo.packagist.org/p2/symfony/options-resolver.json if modified
[304] https://repo.packagist.org/p2/symfony/polyfill-php72.json
[304] https://repo.packagist.org/p2/symfony/polyfill-php80.json
Downloading https://repo.packagist.org/p2/symfony/property-access.json if modified
Downloading https://repo.packagist.org/p2/symfony/property-info.json if modified
[404] https://repo.packagist.org/p2/php-http/client-implementation.json

Fatal error: Declaration of React\Promise\RejectedPromise::then(?callable $onFulfilled = NULL, ?callable $onRejected = NULL, ?callable $onProgress = NULL) must be compatible with React\Promise\PromiseInterface::then($fulfilledHandler = NULL, $errorHandler = NULL, $progressHandler = NULL) in /home/docker/services/project/vendor/react/promise/src/RejectedPromise.php on line 21
```

Composer version 2.2.7 2022-02-25 11:12:27

**Comment by Seldaek** [comment]
Using Composer snapshots (composer self-update --snapshot) should fix this as it ships with a more modern react/promise built-in.

**Comment by cronlabspl** [comment]
@Seldaek why will you not just fix it instead of providing workaround?

**Comment by Seldaek** [comment]
Because it cannot be fixed in older versions (2.2) due to our PHP version support. 2.3 will bump the minimum supported PHP and thus can fix this. Using the prerelease version is a "workaround" if you will, until we release it hopefully soon.

**Comment by cronlabspl** [comment]
@Seldaek 

> Because it cannot be fixed in older versions (2.2) due to our PHP version support.

Seems reasonable :)

> we release it hopefully soon.

Is there an ETA set?

**Comment by Seldaek** [comment]
2.3 is out btw.
